flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -199,6 +200,14 @@ public int getNumberFreeSlotsOf(InstanceID instanceId) {
 	}
 
 	@Override
+	public Map<WorkerResourceSpec, Integer> getPendingWorkerNums() {
+		final int pendingWorkerNum = (int) Math.ceil((double) pendingSlots.size() / numSlotsPerWorker);
```

Comment:
If we want to get rid of back and forth casting and double calculations:
```suggestion
		final int pendingWorkerNum = (pendingSlots.size() - 1) / numSlotsPerWorker + 1;
```

Reply:
On the other hand, it complicates the expression a bit. Maybe introducing a name for this function `multipleOfAndLargerOrEqualThan(numSlotsPerWorker, pendingSlots.size())`.